exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -587,4 +593,77 @@ private void checkStoppedService(Integer serviceId) {
   Optional<ServiceWrapper> findService(String name) {
     return Optional.ofNullable(services.get(name));
   }
+
+  /**
+   * Returns migration script for the given artifact to perform asynchronous data migration.
+   *
+   * @param artifactId Java service artifact id
+   * @param dataVersion base data version migrate from
+   * @return migration script instance or {@link Optional#empty()} if there is no scripts found
+   * @throws IllegalArgumentException if the provided artifact is not deployed
+   * @throws IllegalStateException if there is scripts incompatibility
```

Comment:
if a script is incompatible ? Or `scripts` should be singular.
```suggestion
   * @throws IllegalStateException if there is script incompatibility
```